src/Controller/UsersController.php

```diff
@@ -90,6 +90,13 @@ public function afterIdentify(Event $event, $user)
      */
     public function login()
     {
+        if (isset($this->request->data['username'])) {
+            if (preg_match('/\b[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}\b/i', $this->request->data['username'])) {
+                $this->Flash->error('Invalid username or password, try again.');
```

Comment:
Looks good, but if you don't mind - can you change this error to be/include a note that logins are usernames not emails.

The problem that we're trying to avoid here is that we get new members that try to log in with their email, and they only get an invalid name/ password back which confuses them.

Reply:
We might also want to add a extra hint/ note to the login page itself for the field